### PR TITLE
Add simple-linked-list exercise

### PR DESCRIPTION
**Pull request by vaeng**
The #12in23 challenge asks to do the linked-list exercise, which is currently not implemented and should be added during the month of March. As a preliminary task, we include a simpler version that can be built upon.

- [x] Basic file system
- [x] Write tests 
- [x] Write exercise stub
- [x] Write an example solution
- [x] Write optional documentation files

### CODE REVIEW

**Comment by vaeng** [comment]
I wonder how we can test that the student made sure that no resources have leaked.

**Comment by vaeng** [comment]
I would like to merge this soon, so the users can get their March badges. Would you have a look @siebenschlaefer or @exercism/maintainers-admin

File: config.json [hunk 1]
```diff
@@ -1,848 +1,863 @@
-{
```
Comment (vaeng) [inline config.json:1]: The large change here comes from the configlet format.

File: config.json [hunk 2]
```diff
@@ -1,848 +1,863 @@
...
+        ]
+      },
+      {
+        "slug": "simple-linked-list",
+        "name": "Simple linked list",
+        "uuid": "ffccabe9-9779-4914-9cff-c6f5696c8afe",
+        "practices": ["pointers"],
+        "prerequisites":[],
+        "difficulty": 4,
+        "status": "wip",
```
Comment (vaeng) [inline config.json:790]: Should we publish with `wip` or `beta`?

Comment (ErikSchierboom) [reply config.json:790]: Students won't see the exercise with `wip`.

**Review by ErikSchierboom — COMMENTED** [review]
Thanks a ton for adding this! I'll leave the C++ review to @siebenschlaefer.

File: config.json (hunk 1 above)
Comment (vaeng) [reply config.json:1]: Then I should reverse that to keep the commit small.

File: config.json (hunk 2 above)
Comment (vaeng) [reply config.json:790]: That is how I understood that. Should we publish as beta, so more people can see it? Or should we keep it to the (cpp?) maintainers first? that is what wip would do, right?

File: config.json (hunk 1 above)
Comment (vaeng) [reply config.json:1]: Strange. I cannot see the changes in my IDE. Even with a revert to main.

File: config.json (hunk 2 above)
Comment (ErikSchierboom) [reply config.json]: I think it is fine to have this as `beta`

**Comment by ErikSchierboom** [comment]
I'd like to get this PR merged sooner than later, as this is one of the recommended exercises for Mechanical March

@siebenschlaefer do you have time to do a review of this PR?
@vaeng could you look into fixing the diff for the `config.json` file to not include any formatting changes?

**Comment by vaeng** [comment]
> I'd like to get this PR merged sooner than later, as this is one of the recommended exercises for Mechanical March
> 
> @vaeng could you look into fixing the diff for the `config.json` file to not include any formatting changes?

I had to change the end of line sequence. I uploaded with CRLF but it was LF.

**Review by ErikSchierboom — APPROVED** [review]
LGTM

**Comment by siebenschlaefer** [comment]
I'm late to the party, sorry.

I have one tiny suggestion: Use `std::size_t` instead of `size_t`. The latter will work on all major compilers but technically that's implementation-specific. After including `<cstddef>` the standard only guarantees the existence of `std::size_t`.

**Pull request merged**